test/jdk/java/net/URLClassLoader/getresourceasstream/TestDriver.java

```diff
@@ -53,33 +52,33 @@ public static void main(String[] args)
             throws Throwable {
 
         Path userDir = Paths.get(System.getProperty("user.dir"));
-        String java = JDKToolFinder.getTestJDKTool("java");
         String basename = userDir.getFileName().toString();
         setup(userDir);
         ProcessBuilder[] tests = new ProcessBuilder[]{
-                new ProcessBuilder(
-                        java, TEST_NAME, "./" + ARCHIVE_NAME
+                ProcessTools.createTestJavaProcessBuilder(
+                        TEST_NAME,
+                        "./" + ARCHIVE_NAME
                 ),
-                new ProcessBuilder(
-                        java, "-cp", ".",
+                ProcessTools.createTestJavaProcessBuilder(
+                        "-cp", ".",
```

Comment:
Would it be possible to change the `createJavaProcessBuilder` method in ProcessTools to include something to check if -cp is already an argument being passed?

maybe something like this?

```
if (!noCP && !args.contains("-cp")) {
            args.add("-cp");
            args.add(System.getProperty("java.class.path"));
        }
```

Reply:
That's not a bad idea. What I would recommend in that case would be to not do anything here - but rather log another issue against `ProcessTools.createTestJavaProcessBuilder`. 
Then proceed with this PR without changing anything...

Reply:
Hello Darragh, Daniel,

As for introducing a new method on `ProcessTools` class, I don't know if it's needed - I feel that the `ProcessTools` class is already getting too complex because of the various similarly named methods. I think it might be better to reuse some of those existing `/test/lib` utility methods to achieve the same, something like:

```java
final List<String> command = new ArrayList<>();
command.add(JDKToolFinder.getJDKTool("java"));
command.addAll(jdk.test.lib.Utils.prependTestJavaOpts("-cp", classpath, className, appArgs));
final ProcessBuilder pb = new ProcessBuilder(command);
final OutputAnalyzer outputAnalyzer = ProcessTools.executeCommand(pb);

```
I haven't tried out this snippet to be sure this works as expected.

I am not suggesting we do this change in this current PR.

Reply:
Hi Jaikiran,

What Darragh suggested was to modify the current method in ProcessTools to *not* add -cp <java.class.path> if the argument it is given already contains -cp. Since the first -cp added by the ProcessTools should be ignored anyway by the compiler (last one wins) - then I think it's a good idea, but to follow up outside of the PR.

Reply:
Sorry Daniel and Darragh, I misunderstood the discussion and thought it was a proposal to add a new method in that class. What you suggest about changing the implementation of that existing method, in a separate PR, sounds fine to me.